From ab363a80113880b517c5b7b44808edc12eeaee00 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 27 Nov 2014 10:09:31 +0800
Subject: [PATCH] pyanaconda/timezone.py: fix set timezone failed

If a default timezone exists which provided by tzdata,
the anaconda set timezone failed.

We refer what tzdata did, it installed default timezone
in do_install.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/timezone.py | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/pyanaconda/timezone.py b/pyanaconda/timezone.py
index fdeb771..b8c4ce0 100644
--- a/pyanaconda/timezone.py
+++ b/pyanaconda/timezone.py
@@ -17,7 +17,6 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-import shutil
 import iutil
 import os
 from flags import flags
@@ -36,12 +35,18 @@ class Timezone:
         fromFile = instPath + "/usr/share/zoneinfo/" + self.tz
 
         if not os.access(fromFile, os.R_OK):
-            log.error("Timezone to be copied (%s) doesn't exist" % fromFile)
+            log.error("Timezone to be linked (%s) doesn't exist" % fromFile)
         else:
             try:
-                shutil.copyfile(fromFile, instPath + "/etc/localtime")
+                if os.path.exists(instPath + "/etc/localtime"):
+                    os.unlink(instPath + "/etc/localtime")
+                os.symlink("/usr/share/zoneinfo/" + self.tz, instPath + "/etc/localtime")
             except EnvironmentError as e:
-                log.error("Error copying timezone (from %s): %s" % (fromFile, e.strerror))
+                log.error("Error linking timezone (from %s): %s" % (fromFile, e.strerror))
+
+        f = open(instPath + "/etc/timezone", "w")
+        f.write('%s\n' % self.tz)
+        f.close()
 
         #f = open(instPath + "/etc/sysconfig/clock", "w")
 
-- 
1.9.1

